plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -613,18 +677,53 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
-    private static LongReadFunction oracleTimestampReadFunction()
+    private static String getOracleBindExpression(int precision)
+    {
+        if (precision == 0) {
+            return "TO_TIMESTAMP(?, 'SYYYY-MM-DD HH24:MI:SS')";
+        }
+        if (precision <= 2) {
+            return "TO_TIMESTAMP(?, 'SYYYY-MM-DD HH24:MI:SS.FF')";
+        }
+        else {
+            return format("TO_TIMESTAMP(?, 'SYYYY-MM-DD HH24:MI:SS.FF%d')", precision);
+        }
```

Comment:
nit: redundant else?

```suggestion
        return format("TO_TIMESTAMP(?, 'SYYYY-MM-DD HH24:MI:SS.FF%d')", precision);
```